core/src/main/scala/org/apache/spark/sql/delta/commands/WriteIntoDelta.scala

```diff
@@ -52,6 +52,11 @@ import org.apache.spark.sql.types.StructType
  * In combination with `Overwrite`, a `replaceWhere` option can be used to transactionally
  * replace data that matches a predicate.
  *
+ * In combination with `Overwrite` dynamic partition overwrite mode (option `partitionOverwriteMode`
+ * set to `dynamic`, or in spark conf `spark.sql.sources.partitionOverwriteMode` set to `dynamic`)
+ * is also supported. However a `replaceWhere` will always take precedent over dynamic partition
```

Comment:
i agree if both are set as options on the write. the scenario that bothers me is if dynamic partition overwrite has been enabled in spark conf (so as a global default) and then user sets replaceWhere on a write and gets an error because the two settings conflict...

Reply:
I see the argument that to run replaceWhere instead of throwing error. But that combination needs to be explicitly handled in the code, and properly tested.

Reply:
updated